File: app/src/test/java/com/mss/imagesearcher/di/RobolectricComponent.java
package com.mss.imagesearcher.di;

import com.mss.imagesearcher.view.detail.DetailActivityTest;
import com.mss.imagesearcher.view.imagelist.ImageListActivityTest;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {MockModule.class})
public interface RobolectricComponent extends AppComponent {

    void inject(DetailActivityTest detailActivityTest);

    void inject(ImageListActivityTest imageListActivityTest);
}

File: app/src/test/java/com/mss/imagesearcher/di/RobolectricComponent.kt
package com.mss.imagesearcher.di

import com.mss.imagesearcher.view.detail.DetailFragmentTest
import com.mss.imagesearcher.view.imagelist.MainActivityTest
import com.mss.imagesearcher.view.info.InfoFragmentTest
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [MockModule::class])
interface RobolectricComponent : AppComponent {

    fun inject(detailActivityTest: DetailFragmentTest)

    fun inject(imageListActivityTest: MainActivityTest)

    fun inject(infoActivityTest: InfoFragmentTest)
}
